### DISCUSSION TITLE
[BUG] Cannot bind parameter 'Credentials'. Used documentation

### DISCUSSION THREAD

**Discussion by 200mpg** [pnp/powershell#4166] [General]
Hi, 

I really need help with this. Cannot find information anywhere and boss is hounding me. You are my only hope

 
### Reporting an Issue or Missing Feature
Secret not passing through connect-pnponline

### Expected behavior 
Please describe what output you expect to see from the PnP PowerShell Cmdlets

Connected to Pnp or similar. See reproduction below

### Actual behavior 
Please describe what you see instead. Please provide samples of output or screenshots.

See preproduction below

### Steps to reproduce behavior
Setup secret store, added secret name and value, configured store for no authentication. Run the below command from the **documentation**:

![image](https://github.com/user-attachments/assets/22615b7f-30f5-478c-91b4-1da1d4d13151)

Receive the below:

![image](https://github.com/user-attachments/assets/1e0f5f28-f4f8-46d8-a1ce-5e27c4b91520)

### What is the version of the Cmdlet module you are running?
(you can retrieve this by executing ```Get-Module -Name "PnP.PowerShell" -ListAvailable```)

2.5.0

### Which operating system/environment are you running PnP PowerShell on?
- [X ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by 200mpg** [comment]
@veronicageek Have you been successful in connecting to pnponline using secret stored in secret vault?

**Reply by veronicageek** [reply]
Yes

![image](https://github.com/user-attachments/assets/36e72760-4b66-4872-8822-0803d4a34914)

**Reply by ToddKlindt** [reply]
What kind of object is your secret? 200mpg's secret is a securestring. The error they're getting is that it can't be converted to a Credential. Is your secret a Credential?

**Reply by 200mpg** [reply]
@veronicageek Thank you for the reply. Per the PnP powershell documentation, the PnP app is registered in our tenant, added the generated secret string to a new secret in a local vault. Calling the secret still throws the below:

![Secret](https://github.com/user-attachments/assets/029b1801-44e9-4bbe-9f87-e297f8afbd8c)

![secret2](https://github.com/user-attachments/assets/f1d0915b-f5c9-4b0d-9167-add5988f4114)

Is you're secret string a secure string?
